### PR TITLE
Added option to set signatures to sign repos

### PR DESCRIPTION
**Pull request by breillyr**

### CODE REVIEW

File: conf/dockpulp.conf [hunk 1]
```diff
@@ -61,3 +61,8 @@ test =
 #prod = 1
 #stage = 1
 #test = 1
+
+# Signature keys allowed when creating signed repos
+# This section is optional and environment independent
+#[signatures]
+#beta = foobar
```
Comment (lcarva) [inline conf/dockpulp.conf]: Would be nice to have a new line here

File: dockpulp/__init__.py [hunk 2]
```diff
@@ -1118,6 +1125,20 @@ def getRepos(self, rids, fields=None):
         return self._post('/pulp/api/v2/repositories/search/',
                           data=json.dumps(data))
 
+    def getSignature(self, sig):
+        """Return a signature key."""
+        try:
+            self.sigs[sig]
+        except AttributeError:
```
Comment (twaugh) [inline dockpulp/__init__.py]: Could just handle this exception in the `return...` block below?

Comment (breillyr) [inline dockpulp/__init__.py]: I want to print out the available keys as part of the exception catch - if self.sigs doesn't exist that's going to cause an attribute error at the print.

File: dockpulp/__init__.py [hunk 3]
```diff
@@ -1118,6 +1125,20 @@ def getRepos(self, rids, fields=None):
...
+            log.error('Signatures not defined in dockpulp.conf')
+            exit(1)
```
Comment (lcarva) [inline dockpulp/__init__.py]: How about raising an exception instead of quitting the process?

File: dockpulp/__init__.py (hunk 2 above)
Comment (twaugh) [inline dockpulp/__init__.py]: I just wondered why the code wasn't written like:
```
try:
    return self.sigs[sig]
except AttributeError:
    # handle AttributeError...
except KeyError:
    # handle KeyError...
```

Comment (breillyr) [inline dockpulp/__init__.py]: Oh! I wasn't actually aware you could write try-except blocks that way...good to know for the future.

**Pull request merged**